server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java

```diff
@@ -314,6 +317,31 @@ protected void writeFacets(Request request, QueryContext context, Result<?> resu
     addMandatoryFacetValues(results, RuleIndex.FACET_STATUSES, RuleIndex.ALL_STATUSES_EXCEPT_REMOVED);
     addMandatoryFacetValues(results, RuleIndex.FACET_SEVERITIES, Severity.ALL);
     addMandatoryFacetValues(results, RuleIndex.FACET_TAGS, request.paramAsStrings(PARAM_TAGS));
+
+    mergeNoneAndEmptyBucketOnCharacteristics(results);
+
     super.writeFacets(request, context, results, json);
   }
+
+  protected void mergeNoneAndEmptyBucketOnCharacteristics(Result<?> results) {
+    if (results.getFacets().containsKey(RuleIndex.FACET_DEBT_CHARACTERISTICS)) {
+      Collection<FacetValue> characValues = results.getFacetValues(RuleIndex.FACET_DEBT_CHARACTERISTICS);
+      if (characValues == null) {
+        return;
+      }
+
+      long mergedCount = 0L;
+      Iterator<FacetValue> characIterator = characValues.iterator();
+      while (characIterator.hasNext()) {
+        FacetValue characValue = characIterator.next();
+        if ("".equals(characValue.getKey()) || DebtCharacteristic.NONE.equals(characValue.getKey())) {
```

Comment:
No, it's just that ``""`` is the actual expected value for the ``missing`` aggregation :)